scripts/prove_rpc.sh

```diff
@@ -108,49 +93,76 @@ fi
 # other non-proving code.
 if [[ $8 == "test_only" ]]; then
     # test only run
-    echo "Proving blocks ${BLOCK_INTERVAL} in a test_only mode now... (Total: ${TOT_BLOCKS})"
-    command='cargo r --release --package zero --bin leader -- --test-only --runtime in-memory --load-strategy on-demand --proof-output-dir $PROOF_OUTPUT_DIR --block-batch-size $BLOCK_BATCH_SIZE rpc --rpc-type "$NODE_RPC_TYPE" --rpc-url "$NODE_RPC_URL" --block-interval $BLOCK_INTERVAL  $PREV_PROOF_EXTRA_ARG --backoff "$BACKOFF" --max-retries "$RETRIES" '
+    echo "Proving blocks from ($START_BLOCK) to ($END_BLOCK)"
+    command="cargo r --release --package zero --bin leader -- \
+--test-only \
+--runtime in-memory \
+--load-strategy on-demand \
+--proof-output-dir $PROOF_OUTPUT_DIR \
+--block-batch-size $BLOCK_BATCH_SIZE \
+rpc \
+--rpc-type $NODE_RPC_TYPE \
+--rpc-url $NODE_RPC_URL \
+--start-block $START_BLOCK \
+--end-block $END_BLOCK \
+--backoff $BACKOFF \
+--max-retries $RETRIES \
+$PREV_PROOF_EXTRA_ARG"
+
     if [ "$OUTPUT_TO_TERMINAL" = true ]; then
-        eval $command
+        eval "$command"
         retVal=$?
         echo -e "Proof witness generation finished with result: $retVal"
         exit $retVal
     else
-        eval $command > $OUT_LOG_PATH 2>&1
-        if grep -q 'All proof witnesses have been generated successfully.' $OUT_LOG_PATH; then
+        eval "$command" > "$OUT_LOG_PATH" 2>&1
+        if grep -q 'All proof witnesses have been generated successfully.' "$OUT_LOG_PATH"; then
             echo -e "Success - Note this was just a test, not a proof"
             # Remove the log on success if we don't want to keep it.
             if [ $ALWAYS_WRITE_LOGS -ne 1 ]; then
-                rm $OUT_LOG_PATH
+                rm "$OUT_LOG_PATH"
```

Comment:
Again why double quotation marks? Not to repeat question also on all the other places where they are added in this file.

Reply:
This is a good example where globbing could be undesirable - the rm could remove multiple files unintentionally